packages/next/shared/lib/router/router.ts

```diff
@@ -1864,8 +1866,9 @@ export default class Router implements BaseRouter {
     locale: string | undefined
     isPreview: boolean
   }): Promise<PreflightEffect> {
+    const asPathname = parseUrl(options.as).pathname
```

Comment:
Are we only stripping the query and hash values here if so can we use the `pathNoQueryHash` method?

Reply:
done